005-tailwindcss-framework.md

```diff
@@ -0,0 +1,126 @@
+# Tailwind CSS Framework
+
+We should implement the use of [Tailwind CSS Framework](https://tailwindcss.com/) on our user and admin interfaces to help simplify front-end development across our team.
+
+**Discussion:** <https://github.com/DoSomething/rfcs/pull/5>
+
+## Problem
+
+It has become more difficult over time for our tech team to confidently update the styling and design patterns of our front-end user interfaces.
+
+Our pattern library, [Forge](https://github.com/DoSomething/forge), has worked great for building interfaces with the established patterns it provided. However, the workflow required (using NPM link to test changes locally, publishing releases, updating apps to use new release) did not make updates and rapid iteration to designs patterns feasible, causing Forge to be slow with regular updates and improvements. As a result, developers on the team would often end up adding new patterns directly to an application and never migrating them to Forge for use in other applications.
```

Comment:
> in terms of updating and adding newly defined customizations (both formal pixel/font-size/color definitions & overrides, and extracted common components & updates) in Forge 7.0 -- it'll probably still feel a lot more convenient to just implement it within the respective app itself vs the whole workflow required to update Forge!

True, but in terms of _what_ and _when_ these updates happen I believe they would be much less frequent. It's not often that we would need to add a new color or a new font-size, etc. And same kind of goes for components... we'd only really want to include components in Forge that are generic and whose application could be used in different projects which are also less common in our projects.

Reply:
Yeah, I agree that this is a good step forward but doesn't fully solve that problem. At some point, I'd like to experiment with sharing code between apps [via a monorepo](https://docs.google.com/document/d/1a3e0Wpv9qeRscspdbN4wVR4Chz5oJFb9HH69x2pngSg/edit) so that we don't have to worry about independent versioning for these shared resources.